datafusion/sqllogictest/test_files/case.slt

```diff
@@ -50,3 +50,19 @@ SELECT CASE WHEN a > 2 THEN b END FROM foo
 NULL
 4
 6
+
+# scalar or scalar (string)
```

Comment:
Could you also add a test where both arguments are scalars (like `CASE WHEN 1 > 2 THEN 'true' ELSE 'false')` ?